specifications/transactions/travel-rule/README.md

```diff
@@ -0,0 +1,372 @@
+
+**Final States and next Writer.** The VASP that is the sender of the Payment creates the first `PaymentObject`/`PaymentCommand `, and then the receiver and sender take turns to issue commands updating it until the object they mutate is in one of the final states, namely SABORT, RABORT or READY.
+
+**Protocol Flow Illustration & VASP logic**. Below is a high level illustration of the commands and internal processes a sender and receiver VASPs to define a payment, and update it until it reaches a final state. Light blue blocks represent successful PaymentCommand requests and responses that create or update the PaymentObject. Each is labelled from 1-9, and we will reference these labels when discussing each step in details below.
+
+![](flow.png)
+
+### Steps of the protocol: The KYC Exchange Flow
+
+The Travel Rule Exchange starts with the Sender sending KYC information and requesting KYC information from the receiver. In the simplest approach, this satisfies the Receiver's requiements and the receiver in turn responds with the appropriate KYC information and a recipient signature The Sender would then finalize the payment (by setting its status to `ready_for_settlement`, see below) and then settle on-chain (status sequence SINIT -> RSEND -> READY). At each step Sender or Receiver may also abort the payment, in their turn, until the payment is finalized (READY).
+
+**Start -> SINIT (Step 1)**
+
+The Sender issues `PaymentCommand` to create the initial `PaymentObject`.
+
+The sender creates a `PaymentObject` that they believe requires KYC information exchange. The payment command includes a full `PaymentObject` including all mandatory fields and the following optional fields populated as:
+
+* `sender.status.status` = `need_kyc_data`.
+* `sender.kyc_data` = The `KycDataObject` representing the sender.
+* `receiver.status.status` = `none`
+
+**SINIT -> RSEND (Step 4)**
+
+The Receiver issues `PaymentCommand` to update an existing payment.
+
+The receiver VASP examines the `sender.kyc_data` object, and is satisfied that given the sender information the payment can proceed. It responds with a Payment Command that includes:
+
+* `receiver.status.status` = `ready_for_settlement`
+* `receiver.kyc_data` = The `KycDataObject` representing the receiver.
+* `recipient_signature` = a valid recipient signature.
+
+**SINIT -> RABORT (Step 9)**
+
+The Receiver issues `PaymentCommand` to update an existing payment.
+
+The receiver VASP examines the sender KYC information and is either not satisfied the payment can proceed, needs more time to process the KYC information, or requires additional information to determine if the payment can proceed. It includes a command to abort the payment with an appropriate error code.
+
+* `receiver.status.status`: `abort`
+* `receiver.status.abort_code`: one of `no-kyc-needed` or `rejected`.
+
+The sender can initiate a payment on-chain in case of an abort with `no-kyc-needed`.
```

Comment:
Is it considered a bug from the sender side that the abort_code is no-kyc-needed?

Shouldn't the sender always know whether kyc is needed or does it depend on the receiver?

Reply:
great question, I think there are a lot of edge cases that the current frameworks don't resolve, namely, what happens if I request on an amount below TR or how to handle TR requirements for different regions. There's a lot of work that needs to be done and I suspect we'll likely find ourselves in a situation where we need a lightweight pre-flight before any transaction just to determine if we actually need KYC.